force-app/main/default/lwc/communityBreadCrumbV2/communityBreadCrumbV2.js

```diff
@@ -103,31 +103,44 @@ export default class CommunityBreadCrumbV2 extends LightningElement {
     }
 
     // Retry logic for updating breadcrumbs in case of failure
-    retryBreadcrumbUpdate(attempt = 1, maxAttempts = 5, delay = 500) {
-        if (attempt > maxAttempts) {
-            console.error('Failed to update breadcrumbs after maximum attempts.');
-            return;
+    async retryWithDelay(fn, condition, attempt = 1, maxAttempts = 5, delay = 500) {
+        const result = await fn();
+        if (condition(result)) {
+            return result;
+        }
+
+        if (attempt >= maxAttempts) {
+            console.error('Max retry attempts reached.');
+            return null;
         }
 
         // eslint-disable-next-line @lwc/lwc/no-async-operation, @locker/locker/distorted-window-set-timeout
-        setTimeout(() => {
-            updateBreadcrumbs(this.breadcrumbs);
-
-            if (!this.areBreadcrumbsSet()) {
-                console.log(`Retrying breadcrumb update. Attempt ${attempt}`);
-                this.retryBreadcrumbUpdate(attempt + 1, maxAttempts, delay * 2);
-            } else {
-                console.log('Breadcrumbs set successfully.');
-            }
-        }, delay);
+        await new Promise((resolve) => setTimeout(resolve, delay));
+
+        return this.retryWithDelay(fn, condition, maxAttempts, delay, attempt + 1);
```

Comment:
Her er vel parameterne på feil plass?
Riktig:
`return this.retryWithDelay(fn, condition, attempt + 1, maxAttempts, delay);`

Reply:
Tror forresten det hadde vært mer lesbart med en enkel loop enn rekursjon her:
```
 async retryWithDelay(fn, condition, maxAttempts = 5, delay = 500) {
        let attempt = 1;
        
        while (attempt <= maxAttempts) {
            const result = await fn();
            if (condition(result)) {
                return result;
            }

            if (attempt === maxAttempts) {
                console.error('Max retry attempts reached.');
                return null;
            }

            // eslint-disable-next-line @lwc/lwc/no-async-operation, @locker/locker/distorted-window-set-timeout
            await new Promise((resolve) => setTimeout(resolve, delay));
            attempt++;
        }
    }
```

Reply:
Stemmer, men jeg får en error fra eslint: Unexpected `await` inside a loop.
Er dette farlig eller kan ignoreres?

Reply:
Nei, det går bra